src/cmd/sign.js

```diff
@@ -125,6 +156,170 @@ export default function sign(
   );
 }
 
+/**
+ * Fetches an updateManifest.json file from a remote web server
+ */
+function fetchUpdateManifest(manifestData: Object): Object {
+  return new Promise(async function (resolve, /*reject*/) {
+    let oldUpdateManifest;
+    let statusCode;
+
+    let parsed = url.parse(manifestData.applications.gecko.update_url);
+    let updateManifestFileName = '';
+    // these will never be null, since parsed and parsed.pathname
+    // are already validated in an earlier function
+    // the `if` is only to please flow
+    if (parsed != null && parsed.pathname != null) {
+      updateManifestFileName = path.basename(parsed.pathname);
+    }
+
+    try {
+      [oldUpdateManifest, statusCode] = await httpFetchFile(
+        manifestData.applications.gecko.update_url);
+      if (statusCode < 200 || statusCode > 290) {
+        throw new WebExtError(
+          `Failed to retrieve ${updateManifestFileName}\n` +
+          `http statusCode error, server responded with: ${statusCode}`);
+      }
+    } catch (e) {
+      throw new WebExtError(
+        `Was unable to download ${updateManifestFileName}\nerror is: ${e}`);
+    }
+
+    try {
+      resolve(JSON.parse(oldUpdateManifest));
```

Comment:
Why do you need `resolve()` here? I think if you declare `fetchUpdateManifest` as `async` then you can simply do `return JSON.parse(oldUpdateManifest)` on this line.